packages/api-plugin-simple-schema/src/queries/introspectSchema.js

```diff
@@ -0,0 +1,75 @@
+import SimpleSchema, { toJsonSchema } from "simpl-schema";
+
+const currentPath = ["$"];
+const inputSchema = new SimpleSchema({
+  shopId: String,
+  schemaName: String
+});
+
+/**
+ * @name updateJsonSchema
+ * @summary Update the JSON Schema to add additional path field for each object
+ * @param {Object} jsonSchema - JSON Schema object
+ * @returns {Object} JSON Schema object with path field added
+ */
+function updateJsonSchema(jsonSchema) {
```

Comment:
Since the `currentPath` array is in top-level code and this mutates that array, I worry that this may have unexpected results when run multiple times. It would be better to use an argument with a default value:

```
function updateJsonSchema(jsonSchema, currentPath = ["$"]) {
}
```

And then pass it in wherever you call it recursively:

```js
updateJsonSchema(currentObject[key], currentPath); // recursive call
```

That way it will be reset every time this is called from the query function.

Reply:
Thank you. Made the change and will push soon